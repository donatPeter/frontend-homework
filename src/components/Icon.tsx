import styled from "styled-components";

const StyledIcon = styled.i<{ $size: string }>`
  font-size: ${({ $size }) => $size};
`;

type IconVariant = "outlined" | "filled";

interface Props {
  icon: string;
  size: string;
  variant?: IconVariant;
}

export const Icon = ({ icon, size, variant = "outlined" }: Props) => (
  <StyledIcon
    className={
      variant === "outlined" ? "material-icons-outlined" : "material-icons"
    }
    $size={size}
  >
    {icon}
  </StyledIcon>
);
