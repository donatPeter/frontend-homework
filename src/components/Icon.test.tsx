import { shallow } from "enzyme";
import { Icon } from "./Icon";

describe("<Icon />", () => {
  it("renders icon", () => {
    expect(shallow(<Icon icon="done" size="20px" />)).toMatchSnapshot();
  });
});
