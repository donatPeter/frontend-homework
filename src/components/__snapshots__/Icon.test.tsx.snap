// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders icon 1`] = `
<styled.i
  $size="20px"
  className="material-icons-outlined"
>
  done
</styled.i>
`;
